deployment/src/main/java/io/quarkiverse/quarkus/reactive/h2/client/deployment/ReactiveH2ClientProcessor.java

```diff
@@ -65,13 +65,9 @@ ServiceStartBuildItem build(BuildProducer<FeatureBuildItem> feature,
 
         feature.produce(new FeatureBuildItem(FEATURE));
 
-        createPoolIfDefined(recorder, vertx, eventLoopCount, shutdown, h2Pool, vertxPool, syntheticBeans,
-                DataSourceUtil.DEFAULT_DATASOURCE_NAME, dataSourcesBuildTimeConfig,
-                dataSourcesRuntimeConfig, dataSourcesReactiveBuildTimeConfig, dataSourcesReactiveRuntimeConfig,
-                dataSourcesReactiveH2Config, defaultDataSourceDbKindBuildItems, curateOutcomeBuildItem);
-
-        for (String dataSourceName : dataSourcesBuildTimeConfig.namedDataSources.keySet()) {
-            createPoolIfDefined(recorder, vertx, eventLoopCount, shutdown, h2Pool, vertxPool, syntheticBeans, dataSourceName,
+        for (String dataSourceName : dataSourcesBuildTimeConfig.dataSources().keySet()) {
```

Comment:
The 'dataSourcesBuildTimeConfig.dataSources()' contains the DEFAULT_DATASOURCE_NAME by default, in case no datasource is defined. So, I removed the previous code:

`createPoolIfDefined(recorder, vertx, eventLoopCount, shutdown, h2Pool, vertxPool, syntheticBeans,
                DataSourceUtil.DEFAULT_DATASOURCE_NAME, dataSourcesBuildTimeConfig,
                dataSourcesRuntimeConfig, dataSourcesReactiveBuildTimeConfig, dataSourcesReactiveRuntimeConfig,
                dataSourcesReactiveH2Config, defaultDataSourceDbKindBuildItems, curateOutcomeBuildItem);`